tools/pipelines/test-real-service.yml

```diff
@@ -24,6 +24,7 @@ variables:
 - name: testWorkspace
   value: $(Pipeline.Workspace)/test
 
+lockBehavior: sequential
```

Comment:
should this be set per stage, rather than per pipeline, so that once the stage is complete, other pipelines can make progress: 
https://learn.microsoft.com/en-us/azure/devops/pipelines/process/approvals?view=azure-devops&preserve-view=true&tabs=check-pass#exclusive-lock

Reply:
This is setting the behavior of the lock (which we want for all of them).  Each stage has it's own independent lock, so they don't interfere with each other.

Reply:
the docs are not very good, and i'm not really following how the lock and the variable group interact